src/bcs/INSADMomentumTurbulentNoBCBC.C

```diff
@@ -0,0 +1,35 @@
+#include "INSADMomentumTurbulentNoBCBC.h"
+
+registerMooseObject("MoltresApp", INSADMomentumTurbulentNoBCBC);
+
+InputParameters
+INSADMomentumTurbulentNoBCBC::validParams()
+{
+  InputParameters params = INSADMomentumNoBCBC::validParams();
+  params.addClassDescription("This class implements the 'No BC' boundary condition with "
+                             "additional contributions from turbulent viscosity.");
+  params.addRequiredCoupledVar("mu_tilde", "Spalart-Allmaras turbulence viscosity variable");
+  return params;
+}
+
+INSADMomentumTurbulentNoBCBC::INSADMomentumTurbulentNoBCBC(const InputParameters & parameters)
+  : INSADMomentumNoBCBC(parameters),
+    _mu_tilde(adCoupledValue("mu_tilde"))
+{
+  if (_form == "laplace")
```

Comment:
Yes, `_form` is a Enum type variable that only accepts "laplace" or "traction" in [`INSADObjectTracker`](https://github.com/idaholab/moose/blob/next/modules/navier_stokes/src/userobjects/INSADObjectTracker.C). I built the SA model on top of the existing INSAD (Incompressible Navier-Stokes Automatic Differentiation) model in MOOSE.

Reply:
I guess I was wondering if users *mis*identified their functional form as "traction" when it's actually "laplace" -- this could be ignorance on my part of how MOOSE works. Does the `_form` parameter define the equations used later or does it tell MOOSE how to treat equations specified by the user. If it's the latter case, I can see an issue where this check fails because the specified `_form` and the *actual* form do not match. Unless it is checked somewhere.

Reply:
It's the former case. The "laplace" and "traction" are fundamentally different forms of the Navier-Stokes equations derived using different sets of approximations from the original Navier-Stokes equation (e.g., whether viscosity is constant).